source/request-as-event-emitter.js

```diff
@@ -148,8 +146,7 @@ module.exports = (options = {}) => {
 					total: uploadBodySize
 				});
 
-				const socket = req.connection;
-				if (socket) {
+				req.once('socket', socket => {
```

Comment:
Incorrect implementation here. See why:

1. 30c39bc1c3042a80ed3f6752677e0ab42cb1ca98
2. 13d6b68ba02413374519f490d75741a427d705ad

Reply:
I've reviewed those issues and do not think this change regresses either of them.

- The problem reported in #427 and fixed in #429 was that `req.connection` may be undefined. I've removed the reference to `req.connection` in favor of subscribing to the `socket` event to get the `connection`. So, this should not regress that issue b/c we don't try to access a property which may be undefined.
- The memory leak fix from #465 is addressed by the `if (socket.connecting)` check. It prevents unbounded event subscriptions with keep-alive is enabled on the agent.

Reply:
I've looked deeply and I think it's safe to do so. Good job!